rustler/src/types/i128.rs

```diff
@@ -0,0 +1,130 @@
+use crate::{Decoder, Encoder, Env, Error, NifResult, Term};
+use std::convert::TryFrom;
+
+const EXTERNAL_TERM_FORMAT_VERSION: u8 = 131;
+const SMALL_BIG_EXT: u8 = 110;
+
+impl Encoder for i128 {
+    fn encode<'a>(&self, env: Env<'a>) -> Term<'a> {
+        if let Ok(int) = i64::try_from(*self) {
+            int.encode(env)
+        } else {
+            let mut etf = [0u8; 4 + 16];
+            etf[0] = EXTERNAL_TERM_FORMAT_VERSION;
+            etf[1] = SMALL_BIG_EXT;
+            etf[2] = 16; // length in bytes
+            if *self < 0 {
+                etf[3] = 1;
+                let bytes = (-self).to_le_bytes();
+                etf[4..].copy_from_slice(&bytes);
+            } else {
+                etf[4..].copy_from_slice(&self.to_le_bytes());
+            }
+            let (term, _) = env.binary_to_term(&etf).unwrap();
+            term
+        }
+    }
+}
+
+impl Encoder for u128 {
+    fn encode<'a>(&self, env: Env<'a>) -> Term<'a> {
+        if let Ok(int) = u64::try_from(*self) {
+            int.encode(env)
+        } else {
+            let mut etf = [0u8; 4 + 16];
+            etf[0] = EXTERNAL_TERM_FORMAT_VERSION;
+            etf[1] = SMALL_BIG_EXT;
+            etf[2] = 16; // length in bytes
+            etf[4..].copy_from_slice(&self.to_le_bytes());
+            let (term, _) = env.binary_to_term(&etf).unwrap();
+            term
+        }
+    }
+}
+
+impl<'a> Decoder<'a> for i128 {
+    fn decode(term: Term<'a>) -> NifResult<Self> {
+        if !term.is_integer() {
+            return Err(Error::BadArg);
+        }
+
+        if let Ok(int) = term.decode::<i64>() {
+            return Ok(int as i128);
+        }
+
+        let input = term.to_binary();
+        let input = input.as_slice();
+        if input.len() < 4 {
+            return Err(Error::BadArg);
+        }
+
+        if input[0] != EXTERNAL_TERM_FORMAT_VERSION {
```

Comment:
Can we use `<` here to avoid breaking code when the ETF version is bumped?

Reply:
The ETF version is never bumped amd all libraries hard code it (e.g. https://github.com/sile/eetf/blob/40d786a9e47d7c104466c94e69df8b03afcca5ca/src/codec.rs#L57).